qbertconfig/Fetcher.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 Platform9 Systems, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy
+# of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import logging
+import openstack
+import sys
+from keystoneauth1.exceptions import MissingRequiredOptions
+
+# Local imports
+from QbertClient import QbertClient
+from Kubeconfig import Kubeconfig
+
+LOG = logging.getLogger(__name__)
+
+
+class Fetcher(object):
+    """ Fetches a Kubeconfig from Qbert """
+    def __init__(self, kubeconfig=None, os_config=None, os_cloud=None):
+        """
+        :param kubeconfig: qbertconfig.Kubeconfig object
+        :param os_config: an openstack.config.cloud_config.CloudConfig object
+        :param os_cloud: an openstack.config.cloud_region.CloudRegion object
+        """
+        self.kubeconfig = kubeconfig
+        self.os_config = os_config
+        self.os_cloud = os_cloud
+
+        if not self.kubeconfig:
+            self.kubeconfig = Kubeconfig()
+        if not self.os_config:
+            # this will decide whether to use env vars, or a clouds.yaml
+            self.os_config = openstack.config.OpenStackConfig()
+        self.qbert_session = self._initialize_qbert_client(os_config, os_cloud)
+
+    def _initialize_qbert_client(self, os_config=None, os_cloud=None):
+        """ From a defined OpenStack Cloud, initialize QbertClient
+
+        If cloud_name is not specified, the default values 'envvars' and 'defaults' will be used.
```

Comment:
we can kill this comment since we never implemented this logic and simply rely on openstackSDK to do this